addons/form/nodes/Validator.gd

```diff
@@ -159,12 +154,22 @@ func validate(
 		predefined_regex_result = true
 
 	##-- user_regex --##
-	if user_regex not in ["", null, ".*"]:
+	if user_regex.compile(custom) == OK and user_regex.is_valid() \
+	and user_regex.get_pattern() not in ["", null, ".*"]:
 		if require_single_match:
 			var matches = user_regex.search_all(subject)
 			if matches != null&&matches.size() == 1&&matches[0].get_string() == subject.strip_edges():
 				return true
 			return predefined_regex_result||bool(behaviour)
+
 		elif user_regex.search(subject) != null:
-				return true
-	return true
+			return true
+		
+		broken_rules["custom"] = 'custom'
```

Comment:
```suggestion
		broken_rules["custom"] = custom
```
I think it makes sense to include the pattern here. Do you agree? I haven't touched this code in a while.